packages/siwe/scaffold/views/w3m-connecting-siwe-view/index.ts

```diff
@@ -69,6 +71,13 @@ export class W3mConnectingSiweView extends LitElement {
   }
 
   // -- Private ------------------------------------------- //
+
+  private onRender() {
+    if (SIWEController.state.session) {
+      ModalController.close()
+    }
+  }
```

Comment:
why do we need to close here?

Reply:
there are two flows in Auth
- new 1-click auth flow that resolves after QR scan
- legacy personal_sign request after session is established

this conditions checks if session is already established with the new connect flow, or it should redirect to the legacy `ConnectSiwe` page